### PR TITLE
New DTOs

### PR DESCRIPTION
**Pull request by fabricepipart**

### CODE REVIEW

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import org.trd.app.teknichrono.model.jpa.Event;
+import org.trd.app.teknichrono.model.jpa.Session;
+
+import javax.persistence.EntityManager;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.ArrayList;
+import java.util.List;
+
+@XmlRootElement
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:11]: I don't think this jaxb annotation is needed for JSON serialization using the official JSON-B spec

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  public EventDTO() {
+  }
+
+  public EventDTO(final Event entity) {
+    if (entity != null) {
+      this.id = entity.id;
+      this.version = entity.getVersion();
+      this.name = entity.getName();
+      this.sessions = new ArrayList<>();
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:27]: The instance is already initialized to an empty list, you can delete that line

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      if (entity.getSessions() != null) {
+        for (Session s : entity.getSessions()) {
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:29]: If you want to play it Java 8 style:
* `entity.getSessions().forEach(s -> sessions.add(new NestedSessionDTO(s)))`
* `entity.getSessions().stream().map(NestedSessionDTO::new).forEach(sessions::add)`
* `sessions = entity.getSessions().stream().map(NestedSessionDTO::new).collect(Collectors.toList())`

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          sessions.add(new NestedSessionDTO(s));
+        }
+      }
+    }
+  }
+
+  @Override
+  public String toString() {
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:37]: Just use string concatenation, there is no loop, the generated bytecode will already use a StringBuilder

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    StringBuilder sb = new StringBuilder();
+    sb.append("[id=" + getId());
+    sb.append(",name=" + name + "]");
+    return sb.toString();
+  }
+
+  //
+  public Event fromDTO(Event entity, EntityManager em) {
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:45]: Shouldn't this method be static?

Comment (loicrouchon) [reply src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:45]: By the way, not sure to understand the purpose of that method

File: src/main/java/org/trd/app/teknichrono/model/dto/LocationDTO.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
+package org.trd.app.teknichrono.model.dto;
```
Comment (loicrouchon) [inline src/main/java/org/trd/app/teknichrono/model/dto/LocationDTO.java:1]: Same remarks than for EventDTO

**Review by loicrouchon — APPROVED** [review]
As discussed offline, this code has been generated and will be replaced by a selma mapper implementation soon.

So merging it.

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java (hunk 1 above)
Comment (fabricepipart) [reply src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:11]: Removed in next PR

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java (hunk 2 above)
Comment (fabricepipart) [reply src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:27]: Removed in next PR

File: src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java (hunk 4 above)
Comment (fabricepipart) [reply src/main/java/org/trd/app/teknichrono/model/dto/EventDTO.java:37]: Removed in next PR

**Pull request merged**